### PR TITLE
[FE-3485] Upload file on drop

### PR DESCRIPTION
**Pull request by xixe**

### CODE REVIEW

**Comment by xixe** [comment]
I rewrote the video function to do revokeObjectURL

File: src/modules/common/components/VideoPlayer/VideoPlayer.tsx [hunk 1]
```diff
@@ -38,19 +37,24 @@ export const VideoPlayer = ({
...
-  const renderSrc = useCallback(() => {
-    if (src) {
-      return <source src={src} key={uid(src)} />;
-    }
-    return null;
-  }, [src]);
+  useEffect(() => {
+    return () => {
+      if (isObjectUrl) {
+        URL.revokeObjectURL(video);
```
Comment (BigBadAlien) [inline src/modules/common/components/VideoPlayer/VideoPlayer.tsx:54]: URL.revokeObjectURL instanceof Function

File: src/modules/common/components/VideoPlayer/VideoPlayer.tsx [hunk 2]
```diff
@@ -38,19 +37,24 @@ export const VideoPlayer = ({
...
+  let isObjectUrl = false;
+  let video: string;
+  if (file) {
+    isObjectUrl = true;
+    video = URL.createObjectURL(file);
+  } else if (src) {
+    video = src;
+  } else {
+    throw new Error('No video');
+  }
```
Comment (BigBadAlien) [inline src/modules/common/components/VideoPlayer/VideoPlayer.tsx:49]: const video = useMemo(() => if...)

File: src/modules/common/components/VideoPlayer/VideoPlayer.tsx [hunk 3]
```diff
@@ -38,19 +37,24 @@ export const VideoPlayer = ({
...
 
-  const renderFile = useCallback(() => {
-    if (file) {
-      return <source src={URL.createObjectURL(file)} key={uid(file)} />;
-    }
-    return null;
-  }, [file]);
+  let isObjectUrl = false;
+  let video: string;
+  if (file) {
```
Comment (BigBadAlien) [inline src/modules/common/components/VideoPlayer/VideoPlayer.tsx:42]:
```
{file: File} | {src: string}
```

```
function hasFile(data: any): data is {file: File}  {
  if (data.file) {
     return true;
  }
  
  return false;
}
```


```
if (hasFile(data)) {
   console.log(data.file); // ok
}
```

**Review by BigBadAlien — APPROVED** [review]

**Pull request merged**